### PR TITLE
Move CSP-Headers to middleware

### PR DESCRIPTION
**Pull request by fraxachun**
## Description

Until this PR we defined the Content-Security-Policy header in next.config.js, which is then created during build. The problem is that (at least in our pipelines) environment variables (like process.env.ADMIN_URL) are not available during the build resulting in empty header values.

This has not turned out as a problem until now, because when using the site embedded in an I-Frame it was protected by the OAuth2-Proxy which strips this header. But since https://github.com/vivid-planet/comet/pull/2554 this will produce an error.

The solution is to send this header in the middleware where we have access to the environment variables. However, this (probably) comes with a speed penalty since the code is executed for every request.

### CODE REVIEW

**Review by thomasdax98 — DISMISSED** [review]

**Comment by nsams** [comment]
Not ideal, I would like to have the middleware as small as possible. But there seems to be no other way to do this.

Maybe, you could extract that to an function (and file)?

**Review by johnnyomair — COMMENTED** [review]
What about  the block preview, which isn't handled by the middleware? Shouldn't we add the headers for it as well? And redirects/rewrites (although I believe we don't need them there)

**Review by nsams — DISMISSED** [review]

**Comment by nsams** [comment]
I will come up with a proposal for more elegantly combining multiple middlewares (in a future PR)

**Review by max-debug022 — APPROVED** [review]

**Comment by nsams** [comment]
> I will come up with a proposal for more elegantly combining multiple middlewares (in a future PR)

https://github.com/vivid-planet/comet/pull/2728

**Review by johnnyomair — APPROVED** [review]

**Review by thomasdax98 — DISMISSED** [review]

File: site/next.config.js [hunk 1]
```diff
@@ -24,7 +24,7 @@ const nextConfig = {
     experimental: {
         optimizePackageImports: ["@comet/cms-site"],
     },
-    // https://nextjs.org/docs/advanced-features/security-headers
+    // https://nextjs.org/docs/advanced-features/security-headers (Content-Security-Policy is set in middleware.ts)
```
Comment (dkarnutsch) [inline site/next.config.js:27]:
```suggestion
    // https://nextjs.org/docs/advanced-features/security-headers (Content-Security-Policy and CORS are set in middleware.ts)
```

**Review by dkarnutsch — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**